File: apps/ai/app/schemas/analysis.py
"""Manuscript-analysis request/response contract (maps 1:1 to the analysis_* DB tables)."""

from __future__ import annotations

from app.schemas.common import CamelModel, Confidence, Severity
from app.schemas.internal import LLMUsage, ProjectContext


class ManuscriptAnalysisRequest(CamelModel):
    job_id: str
    file_url: str
    doc_type: str
    project_context: ProjectContext | None = None


class Metric(CamelModel):
    key: str
    value: int
    confidence: Confidence


class CompositionSegment(CamelModel):
    segment: str
    percentage: float


class HeadingCount(CamelModel):
    level: str
    count: int


class RiskFlag(CamelModel):
    severity: Severity
    title: str
    description: str


class TeamSuggestion(CamelModel):
    role: str
    match_score: Confidence
    rationale: str
    candidate_hint: str | None = None


class AnalysisResult(CamelModel):
    overall_confidence: Confidence
    summary: str
    language: str | None = None
    complexity_score: Confidence
    complexity_label: str | None = None
    estimated_working_days: int | None = None
    metrics: list[Metric] = []
    composition: list[CompositionSegment] = []
    headings: list[HeadingCount] = []
    risks: list[RiskFlag] = []
    suggested_team: list[TeamSuggestion] = []
    prompt_id: str | None = None
    model: str | None = None
    usage: LLMUsage | None = None

File: apps/web/src/types/analysis.ts
/** AI analysis domain types (mirror the api ai/analysis modules). */

export type JobStatus = "QUEUED" | "RUNNING" | "SUCCEEDED" | "FAILED";

export interface AiJob {
  jobId: string;
  projectId: string;
  jobType: string;
  status: JobStatus;
  progressPct: number;
  provider?: string | null;
  model?: string | null;
  errorMessage?: string | null;
  createdAt?: string | null;
  finishedAt?: string | null;
}

export interface AnalysisMetric {
  key: string;
  value: number | null;
  confidence: number | null;
}

export interface AnalysisCompositionSegment {
  segment: string;
  percentage: number | null;
}

export interface AnalysisHeadingCount {
  level: string;
  count: number;
}

export interface AnalysisRisk {
  severity: string;
  title: string;
  description?: string | null;
}

export interface AnalysisTeamSuggestion {
  userId?: string | null;
  role?: string | null;
  matchScore?: number | null;
  rationale?: string | null;
}

export interface AnalysisDetail {
  id: string;
  projectId: string;
  aiJobId: string;
  overallConfidence?: number | null;
  summary?: string | null;
  language?: string | null;
  complexityScore?: number | null;
  complexityLabel?: string | null;
  estimatedWorkingDays?: number | null;
  metrics: AnalysisMetric[];
  composition: AnalysisCompositionSegment[];
  headings: AnalysisHeadingCount[];
  risks: AnalysisRisk[];
  suggestedTeam: AnalysisTeamSuggestion[];
  createdAt: string;
}
